src/main/java/edu/harvard/iq/dataverse/DataFileServiceBean.java

```diff
@@ -1657,4 +1689,29 @@ public Embargo findEmbargo(Long id) {
         DataFile d = find(id);
         return d.getEmbargo();
     }
+    
+    public Long getStorageUsageByCreator(AuthenticatedUser user) {
+        Query query = em.createQuery("SELECT SUM(o.filesize) FROM DataFile o WHERE o.creator.id=:creatorId");
```

Comment:
this makes me think (as you point out) how we decide to define the limits even? by user, by dataset - for example a workaround around this limit would be to create a new account and give it permission?)